store/src/java-test/com/zimbra/cs/service/mail/GetFilterRulesTest.java

```diff
@@ -193,10 +195,11 @@ public void testBug71036_SingleRuleSimpleNestedIfwithMultiConditions02() throws
         expectedSoap += "</filterRules>";
 
         Element rules = response.getOptionalElement(MailConstants.E_FILTER_RULES);
-        //ZimbraLog.filter.info(rules.prettyPrint());
-        //ZimbraLog.filter.info(expectedSoap);
-        Assert.assertEquals(expectedSoap, rules.prettyPrint());
 
+        Diff myDiff = DiffBuilder.compare(Input.fromString(rules.prettyPrint()))
+            .withTest(Input.fromString(expectedSoap))
+            .build();
+        Assert.assertFalse(myDiff.toString(), myDiff.hasDifferences());
     }
```

Comment:
May be you can create a separate XMLUtil class in the java-test util package or create a utility method is any existing utility class. Invoke the utility method from all the tests where you need xml comparison.
You may use the below code snippet if you find it useful -
`    public static void assertXMLEquals(String expected, String actual) {
        XMLUnit.setIgnoreWhitespace(true);
        XMLUnit.setIgnoreAttributeOrder(true);

        DetailedDiff diff;
        try {
            diff = new DetailedDiff(XMLUnit.compareXML(expected, actual));
            List<?> allDifferences = diff.getAllDifferences();
            Assert.assertEquals("Differences found: "+ diff.toString(), 0, allDifferences.size());
        } catch (SAXException e) {
            fail("SAXException in comparing xml");
        } catch (IOException e) {
            fail("IOException in comparing xml");
        }
    }`